### PR TITLE
Add API docs

### LINKED ISSUE #10: Document API methods

### PR DESCRIPTION
**Pull request by m-mujica**
Closes #10

screen shot 2017-12-12 at 12 04 00 pm

### CODE REVIEW

File: doc/add-mutated-by.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+@description Register (one to many) mutation dependencies.
+
+@signature `.addMutatedBy(observable, [key], observableOrDependencyRecord)`
+
+Register a mutation dependency between an observable and optional `key`, to one or
+many observables.
+
+Mutation dependencies happen when an observable sets another observable's value,
+the following example creates two [simple observable](can-simple-observable.html)
```
Comment (chasenlehara) [inline doc/add-mutated-by.md]: `[simple observable](can-simple-observable.html)` should probably be `[can-simple-observable simple observable]`

File: doc/add-mutated-by.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+In order to register this dependency (which is critical for [can-debug] to work
+properly), you can use `.addMutatedBy` like this:
+
+```js
+/* code omitted for brevity */
+
+canReflectDeps.addMutatedBy(two, one);
+```
+
+The previous example uses a shorthand to register a value dependency, it's also
```
Comment (chasenlehara) [inline doc/add-mutated-by.md]: “, it's” -> “. It’s” because it’s a new sentence

File: doc/add-mutated-by.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+possible to register `keyDependencies` by passing a dependency record object instead.
+
+Let's build upon the previous example by creating a [simple map](can-simple-map.html)
```
Comment (chasenlehara) [inline doc/add-mutated-by.md]: Same thing with linking to `can-simple-map`

File: doc/add-mutated-by.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+```js
+/* code omitted for brevity */
+
+canReflectDeps.addMutatedBy(two, {
+	valueDependencies: new Set([one]),
+	keyDependencies: new Map([[me, new Set(["age"])]])
+});
+```
+
+An optional `key` can be passed as a second argument when registering [map like](can-reflect.isMapLike.html) observable mutations, e.g:
```
Comment (chasenlehara) [inline doc/add-mutated-by.md]: Another link that I think should be updated.

File: doc/delete-mutated-by.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+@parent can-reflect-dependencies
+
+@description Unregister (one to many) mutation dependencies.
+
+@signature `.deleteMutatedBy(observable, [key], observableOrDependencyRecord)`
+
+Unregister a mutation dependency between an observable and optional `key`, to one 
+or many observables.
+
+Call this method with the exact same arguments passed to [addMutatedBy](can-reflect-dependencies.addMutatedBy.html) to delete the mutation dependency, make sure to do
```
Comment (chasenlehara) [inline doc/delete-mutated-by.md]: I’m going to stop commenting on these.

**Review by chasenlehara — CHANGES_REQUESTED** [review]
These seem really thorough!

**Comment by m-mujica** [comment]
@chasenlehara all links updated. What do you think?

- [ ] yay
- [ ] nay, I have more feedback

❓

File: doc/can-reflect-dependencies.md [hunk 6]
```diff
@@ -135,8 +135,7 @@ things need to happen:
 	`@@@can.getChangesDependencyRecord` symbol.
 
 CanJS observables make this easier by attaching event handling capabilities through
-[can-event-queue](https://github.com/canjs/can-event-queue) mixins, adding in the
-[value mixin](https://github.com/canjs/can-event-queue/blob/master/value/value.js) 
+[can-event-queue] mixins, adding in the [value mixin](https://github.com/canjs/can-event-queue/blob/master/value/value.js)
```
Comment (chasenlehara) [inline doc/can-reflect-dependencies.md]: I think this link should be `[can-event-queue/value/value value mixin]`

**Review by chasenlehara — APPROVED** [review]
Yay! Added a comment for one more thing. 😊

**Comment by m-mujica** [comment]
@chasenlehara fixed that one, too. It should be good now.

**Pull request merged**